### DISCUSSION TITLE
What new in react-telegram-web-app@2.0 ? 🚀🚀🚀

### DISCUSSION THREAD

**Discussion by vkruglikov** [vkruglikov/react-telegram-web-app#32] [Announcements]
## What's new in @vkruglikov/react-telegram-web-app@2.0? 😃🚀

Hey everyone! 
Recently have released a package with breaking changes.

### WebAppProvider [see in docs](https://github.com/vkruglikov/react-telegram-web-app/blob/master/docs/README.md#webappprovider) 🌐📦
The main idea of updating added the `WebAppProvider` component. This added context and enables us better control components and hooks, and the connection between them.

Now WebAppProvider is unnecessary, but it is the better way to use.
It accepts optional property [`options`](https://github.com/vkruglikov/react-telegram-web-app/blob/master/docs/README.md#options )

`smoothButtonsTransition` - When it is `true`, we try to smooth the transition between showing and hiding buttons. 
This happens when using the button on multiple pages 😊

```jsx
import { WebAppProvider, MainButton, BackButton } from '@vkruglikov/react-telegram-web-app';

<WebAppProvider
  options={{
    smoothButtonsTransition: true,
  }}
>
  <Page>
      <MainButton {...} />
      <BackButton {...} />
  </Page>
  <Page>
      <MainButton {...} />
      <BackButton {...} />
  </Page>
</WebAppProvider>
```
So we save you the extra 'hide' call

![IMG_2882_MP4_AdobeExpress](https://github.com/vkruglikov/react-telegram-web-app/assets/9719024/41125050-e201-46e7-87f9-4764d7618de4)
![IMG_2883_MP4_AdobeExpress](https://github.com/vkruglikov/react-telegram-web-app/assets/9719024/3b14e6b2-417b-43cf-8ab8-989cf7f052b2)

### Added tests 🧪
I covered all components and hooks with tests, so it's much safer now

<img width="430" alt="Снимок экрана 2023-07-23 в 13 37 12" src="https://github.com/vkruglikov/react-telegram-web-app/assets/9719024/953e2123-4f09-4879-b53f-0e3c189d0b4f">

### Fixed incorrect WebApp interfaces

Some hooks like `useExpand` return incorrect types.  It's been fixed 

### Summary
These small changes are a big step in the development of the package. 😊
The full list of changes can be viewed here
https://github.com/vkruglikov/react-telegram-web-app/releases/tag/v2.0.0

Enjoy the new update! 🥳🎉

**Comment by milushov** [comment]
Awesome work, man! Thank you 🔥🔥🔥

**Comment by Nasseratic** [comment]
Amazing work 🎉
